Tests/Editor/SettingsAnalysisTests.cs

```diff
@@ -237,5 +238,62 @@ public void SettingsAnalysis_GraphicsUsingRenderingPath_WithBuiltinRenderPipelin
             GraphicsSettings.defaultRenderPipeline = defaultRenderPipeline;
 #endif
         }
+
+        [Test]
+        [TestCase(FogMode.Exponential)]
+        [TestCase(FogMode.ExponentialSquarred)]
+        [TestCase(FogMode.Linear)]
+        public void SettingsAnalysis_FogStripping_IsEnabled(FogMode fogMode)
+        {
+            var serializedObject = new SerializedObject(GraphicsSettings.GetGraphicsSettings());
+            SerializedProperty fogTypeProperty = null;
+
+            switch (fogMode)
+            {
+                case FogMode.Exponential :
+                    fogTypeProperty = serializedObject.FindProperty("m_FogKeepExp");
+                    fogTypeProperty.boolValue = true;
+                    break;
+                case FogMode.ExponentialSquarred :
+                    fogTypeProperty = serializedObject.FindProperty("m_FogKeepExp2");
+                    fogTypeProperty.boolValue = true;
+                    break;
+                case FogMode.Linear :
+                    fogTypeProperty = serializedObject.FindProperty("m_FogKeepLinear");
+                    fogTypeProperty.boolValue = true;
+                    break;
+            }
+
+            serializedObject.ApplyModifiedProperties();
+            Assert.AreEqual(true, FogModeAnalyzer.IsFogStrippingEnabled(fogMode));
+
```

Comment:
I think we could run the analysis here and check we report the corresponding diagnostic. Let's check the description of the issue too to make sure the fog mode is specified.